functions/register/main.py

```diff
@@ -123,9 +126,13 @@ def _get_pending_registration_code(msisdn: str) -> Optional[str]:
 
 
 def _should_send_sms(msisdn: str) -> bool:
-    registration_entities = _get_registration_entities("msisdn", msisdn, timedelta(minutes=1))
+    registration_entities_last_minute = _get_registration_entities("msisdn", msisdn, timedelta(minutes=1))
+    registration_entities_last_hour = _get_registration_entities("msisdn", msisdn, timedelta(hours=1))
+    registration_entities_last_24_hours = _get_registration_entities("msisdn", msisdn, timedelta(days=1))
 
-    if len(registration_entities) > 0:
+    if len(registration_entities_last_minute) > SEND_SMS_LIMIT_PER_MINUTE \
+            or len(registration_entities_last_hour) > SEND_SMS_LIMIT_PER_HOUR \
+            or len(registration_entities_last_24_hours) > SEND_SMS_LIMIT_PER_24_HOURS:
```

Comment:
Widziałem już problem dotyczący [eventual consistency](https://cloud.google.com/datastore/docs/articles/balancing-strong-and-eventual-consistency-with-google-cloud-datastore) w innych plikach ale tutaj wydaje się warty skomentowania. W przypadku zapytań takich jak w `_get_registration_entities` datastore może nie zwrócić ostatnich rekordów z register. Dokumentacja mówi że w niektórych przypadkach opóźnienie może wynosić nawet kilka minut więc wtedy blokada nie zadziała: https://cloud.google.com/datastore/docs/articles/balancing-strong-and-eventual-consistency-with-google-cloud-datastore#eventual-consistency-on-reading-an-index.

Reply:
Rzeczywiście musimy liczyć się z tym, że nie zawsze będzie to skuteczne. Myślę jednak, że możemy zostawić to ograniczenie i dodać rate limiting.